src/Nox.Generator/Domain/EntityMetaGenerator.cs

```diff
@@ -21,63 +25,69 @@ public void Generate(SourceProductionContext context, NoxSolutionCodeGeneratorSt
+                .Select( t =>  GeneraEntityMetaData(t.Value, codeGeneratorState.Solution) )
                 .ToList();
-
+            
             new TemplateCodeBuilder(context, codeGeneratorState)
                 .WithClassName(entity.Name)
                 .WithFileNamePrefix($"Domain.Meta")
                 .WithObject("entity", entity)
-                .WithObject("propertiesWithSource", propertiesWithSource)
+                .WithObject("entitiesMetaData", entitiesMetaData)
                 .GenerateSourceCodeFromResource("Domain.EntityMeta");
         }
     }
 
-    private string? GenerateStaticTypeOptions(NoxSimpleTypeDefinition typeDef, NoxSolution solution)
+    private static EntityMetaData GeneraEntityMetaData(NoxSimpleTypeDefinition typeDef, NoxSolution solution)
     {
-
         var type = typeDef.Type == NoxType.EntityId ? solution.GetSingleKeyTypeForEntity(typeDef.EntityIdTypeOptions!.Entity) : typeDef.Type;
 
         var typeOptions = $"{type}TypeOptions";
 
         var options = typeof(NoxSimpleTypeDefinition).GetProperty(typeOptions);
 
-        var optionsOutput = string.Empty;
-
+        var optionsProperties = Array.Empty<string>();
         var components = type.GetComponents(typeDef).ToArray();
 
         var inParams = components.Length == 1 
             ? $"{components[0].Value.FullName} value" 
             : $"I{type} value" ;
 
-        string? factoryOutput;
-
+       
         if (options != null)
         {
-            var optionsValue = options.GetValue(typeDef, null);
-
-            if (optionsValue != null)
-            {
-                optionsOutput = "public static " +
-                    optionsValue.ToSourceCode($"{typeDef.Name}TypeOptions {{get; private set;}}") + "\r\n";
+            optionsProperties = GetTypeOptionProperties(typeDef, options);
+        }
+        
+        return new EntityMetaData
+        {
+            Name = typeDef.Name,
+            Type = type.ToString(),
+            InParams = inParams,
+            HasTypeOptions = optionsProperties.Any(),
+            OptionsProperties = optionsProperties
+        };
+    }
 
-                factoryOutput =
-                    $"public static {type} Create{typeDef.Name}({inParams})\r\n" +
-                    $"    => Nox.Types.{type}.From(value, {typeDef.Name}TypeOptions);" +
-                    $"\r\n\r\n";
+    private static string[] GetTypeOptionProperties(
+        NoxSimpleTypeDefinition typeDef, 
+        PropertyInfo options)
+    {
+        var optionsProperties = Array.Empty<string>();
+        var optionsValue = options.GetValue(typeDef, null);
 
-                return optionsOutput + factoryOutput;
-            }
+        if (optionsValue != null)
+        {
+            var output = optionsValue.ToSourceCode($"{typeDef.Name}TypeOptions {{get; private set;}}");
+            var optionsOutputLines = output.Split('\n');
+            optionsProperties = optionsOutputLines
```

Comment:
this is high complex to maintain, can we skip the ToSourceCode and this complexity an create a method that gathers the info we need so in the template we can generate the code that can be easily maintained?

Teh ObjectExtensions should not generate any code and just gather the info in the structured way and the template generates the code

Reply:
lets solve this on a second PR